torchaudio/datasets/dr_vctk.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from typing import Dict, Tuple, Union
+
+from torch import Tensor
+from torch.utils.data import Dataset
+
+import torchaudio
+from torchaudio.datasets.utils import (
+    download_url,
+    extract_archive,
+)
+
+
+URL = "https://datashare.ed.ac.uk/bitstream/handle/10283/3038/DR-VCTK.zip"
+FOLDER_IN_ARCHIVE = Path("DR-VCTK") / "DR-VCTK"
+_CHECKSUMS = {
+    "https://datashare.ed.ac.uk/bitstream/handle/10283/3038/DR-VCTK.zip": "29e93debeb0e779986542229a81ff29b",
+}
+
+
+class DR_VCTK(Dataset):
+    """Create a dataset for Device Recorded VCTK (Small subset version).
+
+    Args:
+        root (str or Path): Root directory where the dataset's top level directory is found.
+        download (bool):
+            Whether to download the dataset if it is not found at root path. (default: ``False``).
+        url (str): The URL to download the dataset from.
+            (default: ``"https://datashare.ed.ac.uk/bitstream/handle/10283/3038/DR-VCTK.zip"``)
+        train (bool): If True, creates dataset from training set, otherwise creates from test set.
+            (default: ``True``).
+    """
+
+    def __init__(
+        self,
+        root: Union[str, Path],
+        download: bool = False,
+        url: str = URL,
+        folder_in_archive: Union[str, Path] = FOLDER_IN_ARCHIVE,
+        train: bool = True,
+    ) -> None:
+        root = Path(root).expanduser()
+        folder_in_archive = Path(folder_in_archive)
+
+        archive = root / "DR-VCTK.zip"
+
+        self._subset = "train" if train else "test"
+        self._path = root / folder_in_archive
+        self._clean_audio_dir = self._path / f"clean_{self._subset}set_wav_16k"
+        self._noisy_audio_dir = self._path / f"device-recorded_{self._subset}set_wav_16k"
+        self._config_filepath = self._path / "configurations" / f"{self._subset}_ch_log.txt"
+
+        if not self._path.is_dir():
+            if not archive.is_file():
+                if not download:
+                    raise RuntimeError("Dataset not found. Please use `download=True` to download it.")
+                checksum = _CHECKSUMS.get(url, None)
+                download_url(url, root, hash_value=checksum, hash_type="md5")
```

Comment:
Can you move the verification of checksum before `extract_archive`?
So that the verification is performed even when the code is not downloading but just extracting.